### PR TITLE
Adding information about the scripts for generating test data on readme

### PR DESCRIPTION
**Pull request by era**
This PR does two things:

1) While running the scripts, one of them failed because it could not find `../trustfall_core/test_data/schemas/{schema_name}.graphql` the error message was just a plain "could not find file" (`thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }', trustfall_testbin/src/main.rs:44:14`) which forced me to open the file to discover which file was missing. I added a `context` to the error to make this easier in the future. I can remove it if you think is not needed.
2) I wrote a little bit about the scripts to generate test data. I didn't describe each one of them but wrote what they do. Let me know if I understood it wrong tho.

### CODE REVIEW

File: trustfall_core/test_data/README.md [hunk 1]
```diff
@@ -67,3 +67,26 @@ or produce a specified error ([`schema_errors`](./tests/schema_errors/) director
...
+files based on that.
+
+The scripts named `create_*` receive as input a list of `graphql.ron` files and generate 
+all needed files for a given test case. For example, `create_frontend_error_data.sh` will
+create `*.graphql-parsed.ron` and `*.frontend-error.ron` files.
+
+If you want to create a new valid query test, you can run:
+`./create_valid_query_data.sh ../trustfall_core/test_data/tests/valid_queries/my_test.graphql.ron`.
+
+Scripts must be run inside the `scripts` folder.
```
Comment (obi1kenobi) [inline trustfall_core/test_data/README.md:92]: Is this true? I thought I had made them be location-independent. Could you double-check?

If any of them are not location-independent, that's also worth fixing, perhaps in the next PR.

File: trustfall_testbin/src/main.rs [hunk 2]
```diff
@@ -1,6 +1,7 @@
 #![forbid(unsafe_code)]
 #![forbid(unused_lifetimes)]
 
+use anyhow::Context;
```
Comment (obi1kenobi) [inline trustfall_testbin/src/main.rs:4]:
```suggestion
use anyhow::Context as _;
```
This is a cute little trick: it imports the trait (making its methods visible) but without importing the name `Context`.

**Review by obi1kenobi — COMMENTED** [review]
Broadly looks good! Just a couple of minor things to take a quick peek at before I merge.

File: trustfall_testbin/src/main.rs (hunk 2 above)
Comment (era) [reply trustfall_testbin/src/main.rs:4]: TIL

File: trustfall_core/test_data/README.md (hunk 1 above)
Comment (era) [reply trustfall_core/test_data/README.md:92]: Yes, not sure if there are more things to change, but the `get_schema_by_name` hardcodes the path of the schema to be `../trustfall_core/test_data/schemas/{schema_name}.graphql`, which fails if you run the script from the repo root:

```
λ ~/programming/trustfall/ era/projection* ./scripts/create_valid_query_data.sh trustfall_core/test_data/tests/valid_queries/elias_test.graphql.ron
> Starting on file trustfall_core/test_data/tests/valid_queries/elias_test.graphql.ron
   Compiling trustfall_testbin v0.1.0 (/home/era/programming/trustfall/trustfall_testbin)
    Finished dev [unoptimized + debuginfo] target(s) in 2.52s
     Running `target/debug/trustfall_testbin parse trustfall_core/test_data/tests/valid_queries/elias_test.graphql.ron`
    Finished dev [unoptimized + debuginfo] target(s) in 0.04s
     Running `target/debug/trustfall_testbin frontend trustfall_core/test_data/tests/valid_queries/elias_test.graphql-parsed.ron`
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: failed to read schema from ../trustfall_core/test_data/schemas/numbers.graphql

Caused by:
    No such file or directory (os error 2)', trustfall_testbin/src/main.rs:46:10
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
But it just works if I go to the scripts folder (well, and any other folder that is just 1 level depth in the repo).

I can change in this or the next PR, is up to you :)

Comment (obi1kenobi) [reply trustfall_core/test_data/README.md:92]: Nice catch! I thought `cargo run` imposed a bit more control over the pwd where the program runs. Let's change it in the next PR and merge this.

**Review by obi1kenobi — APPROVED** [review]

**Pull request merged**